### DISCUSSION TITLE
[Voice instructions] don't say ref attribute if the road is not very important

### DISCUSSION THREAD

**Discussion by qwerty287** [osmandapp/OsmAnd#12186] [General]
### Description 
OsmAnd's voice instructions always speaks the [ref attribute](https://wiki.openstreetmap.org/wiki/Key:ref) for roads if it's available. If the road is a highway or another important street ([`highway=motorway`](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dmotorway), [`highway=trunk`](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dtrunk) or  [`highway=primary`](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dprimary) as OSM tag), this is very useful. But if it's something else, you probably can't use these numbers, because they are actually not shown in reality (at least in Germany). As an example: https://www.openstreetmap.org/way/221808444 - OsmAnd would say "K 6422 Eggersdorfer Strasse", which is very long and annoying. So it would be nice if OsmAnd wouldn't say them if the road is not very important (but show it with the street name in the top bar). Maybe, as an exception, it could be used if the road doesn't have another name. As I don't know how this is in other countries, there could be an option to change it (with values like "always", "only important roads" and "never").

Thank you!

**Comment by quax2712** [comment]
This is a very good idea, the ref attribute for tertiary and residential streets is not shown on signs in Germany. If possible, it should be set using new setup options.

**Comment by sonora** [comment]
This is a country by country decision, in e.g. the US the use of highway numbers is very predominant down to the county road level (which is equivalent to the level you report here should be suppressed for Germany).

I am a strong believer in delivering software usefully configurd out of the box. This means that for such a fundamental functionality we should not simply suppress something esential, and burden the user to find a setting and change it.

Hence I suggest the only fix can be we introduce a cut-off level above which refs are conventionally used, and then intrinsically set it per country. Until ee have that, or a similar algorithm,I am afraid we should leave things as they are?

**Comment by vshcherb** [comment]
Converted to a discussion cause the solution is not obvious at all. In the Netherlands I believe Refs for tertiary are very good indication and they are displayed. Then it's a quite tricky setting that depends on the country or probably on the region of the country.

**Comment by qwerty287** [comment]
Yes, it depends on the county, so you oculd enable it by default and just add the option to disable it for specific road types. This would make it working out of the box but if the user don't want it, you could disable it. Actually, a per-country/region solution would be the best option, but it's much effort and not that easy because you would need the data for every country in the world.

**Reply by issteve** [reply]
Yes, for Germany this could be e.g. an option to turn on/off refs like "K####". 

The "L###" I personally found useful because they are sometimes signed with little white signs beside the road so this information gives me the possibility to recheck if I'm on the right road (e.g. if going parallel to another one).

And I would like to point out, for me it's only about the ref "K####" additional to another street name and not if the "K####" is the only street name. If the street has no other name than "K####" I definitely would like to hear it, because I normally check the navigation before going on the road and keep some important steps in mind and would need it as a reminder then.

**Comment by issteve** [comment]
Yes, just don't confuse it with this bug: https://github.com/osmandapp/OsmAnd/issues/12180 
- as sometimes it really shouldn't be said (e.g. when it doesn't exist and would be just the one you are already traveling on instead of the one you should change to.).
